proposed/apphost-embed-install-location.md

```diff
@@ -0,0 +1,158 @@
+- [dotnet/runtime#70975](https://github.com/dotnet/runtime/issues/70975)
+- [dotnet/runtime#86801](https://github.com/dotnet/runtime/issues/86801)
+
+## State in .NET 8
+
+The current state is described in detail in
+[install-locations](https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md)
+and [install-locations-per-architecture](https://github.com/dotnet/designs/blob/main/accepted/2021/install-locations-per-architecture.md)
+
+At a high level, the process and priority for `apphost` determining the install
+location is:
+
+  1. App-local
+      - Look for the runtime in the app's folder (self-contained apps)
+  2. Environment variables
+      - Read the `DOTNET_ROOT_<arch>` and `DOTNET_ROOT` environment variables
+  3. Global install (registered)
+      - Check for a registered install - registry key on Windows or a install
+      location file on non-Windows
+  4. Global install (default)
+      - Fall back to a well-known default install location based on the platform
+
+## Embedded install location options for `apphost`
+
+Every `apphost` already has a placeholder that gets rewritten to contain the app
+binary path during build (that is, `dotnet build`). This proposal adds another
+placeholder which would represent the optional configuration of search locations
+and embedded relative path to an install location. Same as existing placeholder,
+it would conditionally be rewritten during build based on the app's settings.
+This is similar to the proposal in [dotnet/runtime#64430](https://github.com/dotnet/runtime/issues/64430),
+but with additional configuration of which search locations to use.
+
+### Configuration of search locations
+
+Some applications do not want to use all the default search locations when they
+are deployed - for example, [dotnet/runtime#86801](https://github.com/dotnet/runtime/issues/86801).
+
+We can allow selection of which search locations the `apphost` will use. When
+search locations are configured, only the specified locations will be searched.
+
+```c
+enum search_location
+{
+    sl_default = 0,
+    sl_app_local = 1 << 0,
+    sl_embedded = 1 << 1,
+    sl_environment_variables = 1 << 2,
+    sl_global = 1 << 3,
+};
+```
+
+The behaviour corresponding to the `default` value is defined by the `apphost`.
+It is currently `app_local | environment_variables | global`.
+
+The search location could be specified via a property in the project:
+
+```xml
+<AppHostDotNetSearch>Global</AppHostDotNetSearch>
+```
+
+where the valid values are `AppLocal`, `Embedded`, `EnvironmentVariables`, or
+`Global`. Multiple values can be specified, delimited by semi-colons.
+
+When a value is specified, only those locations will be used. For example, if
+`Global` is specified, `apphost` will only look at global install locations, not
+app-local, at any embedded path, or at environment variables.
+
+### Embedded relative path to install location
```

Comment:
Naming: I would prefer we don't use "embedded" as the term here. To me it reads like the "DotNet" is embedded - which originally lead me to believe that this also incorporates single-file behavior.

Given the other values, I think it would make sense to call this `AppRelative`.

Similarly I would not use `AppHostDotNetRoot` - especially the "root" part - the value should only accept relative paths, so we should name it as such.